src/rgw/rgw_lua_utils.h

```diff
@@ -147,59 +159,63 @@ void create_metatable(lua_State* L, bool toplevel, Upvalues... upvalues)
     // duplicate the table to make sure it remain in the stack
     lua_pushvalue(L, -1);
     // give table a name (in cae of "toplevel")
-    lua_setglobal(L, MetaTable::TableName().c_str());
+    lua_setglobal(L, qualified_name.c_str());
   }
-  // create metatable
-  [[maybe_unused]] const auto rc = luaL_newmetatable(L, MetaTable::Name().c_str());
+
+  // create/reuse metatable
+  [[maybe_unused]] const auto metatable_is_new = luaL_newmetatable(L, qualified_name.c_str());
```

Comment:
So I had expected to see some kind of check that the metatable is new. If it is, then set these metafields below.

Again, it's very odd and unconventional to see an existing metatable have its metafields modified.

Reply:
with the fix, the code would either create a new metatable, as in this case:
```lua
local o1 = Request.Object -- new metatable is created to represent the Object in Request.Object 
local o2 = Request.CopyFrom.Object -- new metatable (with different upvalues) is created to represent Request.CopyFrom.Object
print(o1.Name)
print(o2.Name)
```
or, will reuse an existing metatable, as in this case:
```lua
local o1 = Request.Object -- new metatable is created to represent the Object in Request.Object 
local o2 = Request.Object -- reuse the same metatable
print(o1.Name)
print(o2.Name)
```

hmm... maybe i should just do?
```suggestion
  const auto metatable_is_new = luaL_newmetatable(L, qualified_name.c_str());
  if (!metatable_is_new) return;
```

will check that!

Reply:
@batrick you are right, if the metatable already exists, it should only be tied to the new table, no need to assign it with new values. fixed in: https://github.com/ceph/ceph/commit/f52f4918d996566cc4ab9685b7ba1863cff473c0